internal/pkg/otel/manager/manager.go

```diff
@@ -179,11 +238,15 @@ func (m *OTelManager) Run(ctx context.Context) error {
 				// that results in the collector being started
 				proc, err = m.execution.startCollector(ctx, m.logger, m.cfg, collectorRunErr, m.statusCh)
 				if err != nil {
+					m.logger.Errorf("collector exited with error: %v", err)
 					// failed to create the collector (this is different then
 					// it's failing to run). we do not retry creation on failure
 					// as it will always fail. A new configuration is required for
 					// it not to fail (a new configuration will result in the retry)
 					reportErr(ctx, m.errCh, err)
+					// since this is a new configuration we want to start the timer
+					// from the initial delay
+					m.recoveryTimer.ResetInitial()
```

Comment:
Only doing `ResetInitial` on configuration change seems to mean that for rare failures at steady state (think a panic that happens every 1h-1 day) the restart duration will lengthen for each retry. I don't think this is right.

The reason to backoff is to avoid spamming the machine with a process that just exits immediately.  We probably want to reset the backoff after the collector has been healthy enough.

In standalone agents, the configuration may never change after the initial start. In Fleet managed agents, it can be days to weeks between changes. They are not a constant thing that we can rely on as the "clock" of sub-process management.

Reply:
nice point, I did extend the recoveryBackoff to reset to the initial backoff after some time here https://github.com/elastic/elastic-agent/pull/8248/commits/704e25e66b08edfb92646076fe62354730d6ed04. Also added a tests to validate that it actually resets to initial interval 🙂